pkg/resources/requests_test.go

```diff
@@ -121,3 +123,76 @@ func TestCountIn(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceQuantityRoundTrips(t *testing.T) {
+	cases := map[string]struct {
+		resource corev1.ResourceName
+		value    int64
+		expected string
+	}{
+		"1": {
+			resource: corev1.ResourceMemory,
+			value:    1,
+			expected: "1",
+		},
+		"1k": {
+			resource: corev1.ResourceMemory,
+			value:    1000,
+			expected: "1k",
+		},
+		"1M": {
+			resource: corev1.ResourceMemory,
+			value:    1000000,
+			expected: "1M",
+		},
+		"1.5M": {
+			resource: corev1.ResourceMemory,
+			value:    1500000,
+			expected: "1500k",
+		},
```

Comment:
Question, what it means "round-trips" in this context, since the serialization looks different: "1.5M" vs "1500k"?

Reply:
The test case name doesn't participate in the test logic. The initial value is int64 and not a string. I just found the 1.5 string easier to read, in particular for MiB.

I think if we parsed a string "1.5M", it wouldn't roundtrip and would serialize back to "1500k". But it's not a problem, because "1.5M" could only be provided as an input which is then converted to int64 before being sent to the API, and after that conversion it consistently roundtrips.

I updated the test case name